mlflow/fastai/callback.py

```diff
@@ -0,0 +1,156 @@
+            return
+        # Remove non-float metrics and record the rest.
+        metrics = self.recorder.log
+        metrics = {k: v for k, v in zip(self.recorder.metric_names, metrics) if _is_float(v)}
+        self.metrics_logger.record_metrics(metrics, step=metrics["epoch"])
+
+    def before_fit(self):
+        from fastai.callback.all import ParamScheduler
+
+        # Do not record in case of predicting or lr_finder
+        if hasattr(self, "lr_finder") or hasattr(self, "gather_preds"):
+            return
+
+        if self.is_fine_tune and len(self.opt.param_lists) == 1:
+            print(
+                """ WARNING: Using `fine_tune` with model which cannot be frozen.
+                Current model have only one param group which makes it impossible to freeze.
+                Because of this it will record some fitting params twice (overriding exception)
+                """
+            )
+
+        frozen = self.opt.frozen_idx != 0
+        if frozen and self.is_fine_tune:
+            self.freeze_prefix = "freeze_"
+            try_mlflow_log(mlflow.log_param, "frozen_idx", self.opt.frozen_idx)
+        else:
+            self.freeze_prefix = ""
+
+        # Extract function name when `opt_func` is partial function
+        if isinstance(self.opt_func, partial):
+            try_mlflow_log(
+                mlflow.log_param,
+                self.freeze_prefix + "opt_func",
+                self.opt_func.keywords["opt"].__name__,
+            )
+        else:
+            try_mlflow_log(
+                mlflow.log_param, self.freeze_prefix + "opt_func", self.opt_func.__name__
+            )
+
+        params_not_to_log = []
+        for cb in self.cbs:
+            if isinstance(cb, ParamScheduler):
+                params_not_to_log = list(cb.scheds.keys())
+                for param, f in cb.scheds.items():
+                    values = []
+                    for step in np.linspace(0, 1, num=100, endpoint=False):
+                        values.append(f(step))
+                    values = np.array(values)
+
+                    # Log params main values from scheduling
+                    try_mlflow_log(
+                        mlflow.log_param, self.freeze_prefix + param + "_min", np.min(values, 0)
+                    )
+                    try_mlflow_log(
+                        mlflow.log_param, self.freeze_prefix + param + "_max", np.max(values, 0)
+                    )
+                    try_mlflow_log(
+                        mlflow.log_param, self.freeze_prefix + param + "_init", values[0]
+                    )
+                    try_mlflow_log(
+                        mlflow.log_param, self.freeze_prefix + param + "_final", values[-1]
+                    )
+
+                    # Plot and save image of scheduling
+                    fig = plt.figure()
+                    plt.plot(values)
+                    plt.ylabel(param)
```

Comment:
A plot for `freeze_lr` looks like this:

<img width="50%" src="https://user-images.githubusercontent.com/17039389/133072511-ce4af7d2-568c-496b-a1d5-aae44cbf5cb7.png" />

- There is no x-axis label.
- There is no legend.